### PR TITLE
feat: approve only

### LINKED ISSUE #8: GitHub actions workflows are not triggered when PRs are merged by this action
## 🐛 Bug Report

GitHub actions workflows are not triggered when PRs are merged by this action.

See the long discussion in https://github.com/nearform/fastify-secrets-core/issues/59, which is using this action.

The behavior is by design, to avoid recursive workflow triggers.

## Expected behavior

We would like to have workflows triggered by merges done by this action. One of the options is to use a GitHub application to do so. It may be worth researching how to do that.

### PR DESCRIPTION
**Pull request by simoneb**
Because of https://github.com/fastify/github-action-merge-dependabot/issues/8, we need a different way to run builds when dependabot PRs are merged.

GitHub now has an automerge option in the repository settings, which allows merging a PR when requirements are satisfied. Those requirements usually include:

- a green build
- one approval

This feature allows this action to only provide the approval without merging, so that we can use the built-in GH feature to do the merge and run the CI checks instead.

It's clearly not exactly the same thing, because using GH's automerge would automerge other PRs besides the dependabot ones, but it's good enough.

Closes #8

### CODE REVIEW

**Comment by mcollina** [comment]
Seems good for me, however this require per-project specific config which will be highly impractical in this org.

File: README.md [hunk 1]
```diff
@@ -39,23 +42,22 @@ jobs:
       - uses: fastify/github-action-merge-dependabot@v1
         if: ${{ github.actor == 'dependabot[bot]' && github.event_name == 'pull_request' }}
         with:
-          github-token: ${{secrets.github_token}}
+          github-token: ${{secrets.GITHUB_TOKEN}}
```
Comment (mcollina) [inline README.md:45]: why this change?

Comment (simoneb) [reply README.md:45]: because it was wrong https://docs.github.com/en/actions/reference/authentication-in-a-workflow

**Comment by simoneb** [comment]
> Seems good for me, however this require per-project specific config which will be highly impractical in this org.

I agree, but this action is used outside of this org too

**Review by mcollina — APPROVED** [review]
lgtm

File: src/index.js [hunk 2]
```diff
@@ -4,50 +4,67 @@ const github = require('@actions/github')
...
 
-async function run () {
+async function run() {
   try {
     const octokit = github.getOctokit(GITHUB_TOKEN)
 
     const { repository, pull_request: pr } = github.context.payload
+
+    if (!pr) {
+      throw new Error(
```
Comment (salmanm) [inline src/index.js:22]: Shouldn't we silently ignore?

**Review by salmanm — APPROVED** [review]
lgtm

File: src/index.js (hunk 2 above)
Comment (simoneb) [reply src/index.js:22]: we could return. I kept the existing semantics though, just breaking it in a more explicit way. without this it would still break here https://github.com/fastify/github-action-merge-dependabot/pull/9/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R29
